pokemongo_bot/cell_workers/follow_spiral.py

```diff
@@ -79,48 +78,13 @@ def work(self):
             point['lng']
         )
 
-        alt = uniform(self.bot.config.alt_min, self.bot.config.alt_max)
-        if self.bot.config.walk_max > 0:
-            step_walker = StepWalker(
-                self.bot,
-                point['lat'],
-                point['lng']
-            )
-
-            if self.cnt == 1:
-                self.emit_event(
-                    'position_update',
-                    formatted="Walking from {last_position} to {current_position} ({distance} {distance_unit})",
-                    data={
-                        'last_position': (last_lat, last_lng, last_alt),
-                        'current_position': (point['lat'], point['lng'], alt),
-                        'distance': dist,
-                        'distance_unit': 'm'
-                    }
-                )
-
-            if step_walker.step():
-                step_walker = None
-        else:
-            self.bot.api.set_position(point['lat'], point['lng'], alt)
-            self.emit_event(
-                'position_update',
-                formatted="Teleported from {last_position} to {current_position} ({distance} {distance_unit})",
-                data={
-                    'last_position': (last_lat, last_lng, last_alt),
-                    'current_position': (point['lat'], point['lng'], alt),
-                    'distance': dist,
-                    'distance_unit': 'm'
-                }
-            )
-
-        if dist <= 1 or (self.bot.config.walk_min > 0 and step_walker == None):
-            if self.ptr + self.direction >= len(self.points) or self.ptr + self.direction <= -1:
-                self.direction *= -1
```

Comment:
This would be my simplified version of the whole  .work()

```
    def work(self):
        last_lat, last_lng, last_alt = self.bot.position()
        point = self.points[self.ptr]

        step_walker = StepWalker(
            self.bot,
            point['lat'],
            point['lng']
        )
        
        if step_walker.step():
            if len(self.points) > 1:
                self.points.pop(self.ptr)

        return [point['lat'], point['lng']]
```